web/src/engine/websites/CopyManga.ts

```diff
@@ -0,0 +1,161 @@
+import { Tags } from '../Tags';
+import icon from './CopyManga.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Check, Choice, Text } from '../SettingsManager';
+import { EngineResourceKey as E, WebsiteResourceKey as W } from '../../i18n/ILocale';
+import { FetchRequest } from '../FetchProviderNodeWebkit';
+import { FetchJSON } from '../FetchProvider';
+
+type APIResponse<T> = {
+    code: number,
+    message: string,
+    results: T
+}
+
+type APISingleComic = {
+    comic: APIComic
+}
+
+type APIResultList<T> = {
+    list: T[]
+}
+
+type APIPages = {
+    chapter: APIChapter
+}
+
+type APIComic = {
+    uuid: string,
+    path_word: string,
+    name: string
+}
+
+type APIChapter = {
+    uuid: string,
+    name: string,
+    contents: {
+        url: string
+    }[];
+    words: number[]
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly API_HEADERS = {
+        'User-Agent': '"User-Agent" to "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.124 Safari/537.36 Edg/102.0.1245.44"',
```

Comment:
Consider using `"User-Agent" to "${navigator.userAgent}"` in case the user has changed the user-agent (e.g., via HakuNeko Assistant to bypass CloudFlare)

Reply:
to be frank it took this from the connector in hakuneko i dont know if a custom UA is really necessary.